testfixtures/popen.py

```diff
@@ -54,15 +54,43 @@ def set_command(self, command, stdout=b'', stderr=b'', returncode=0,
         """
         self.commands[command] = (stdout, stderr, returncode, pid, poll_count)
 
+    def set_callable(self, command, command_callable):
```

Comment:
I don't think this warrants its own method, feels like it should just be an options to `set_command`, maybe called `side_effect` in a similar vein to Mock's usage.